src/Workspaces/CoreTest/SolutionTests/ProjectDependencyGraphTests.cs

```diff
@@ -397,13 +400,256 @@ public void TestReverseTransitiveReferencesForUnrelatedProjectAfterWithProjectRe
+            Assert.NotNull(notExpected);
+
+            var aToB = a.ProjectReferences.Single(reference => reference.ProjectId == b.Id);
+            solution = solution.RemoveProjectReference(a.Id, aToB);
+
+            // Before any other operations, verify that TryGetProjectsThatTransitivelyDependOnThisProject returns a
+            // null set.
+            Assert.Null(solution.State.GetProjectDependencyGraph().GetTestAccessor().TryGetProjectsThatTransitivelyDependOnThisProject(c.Id));
+            VerifyReverseTransitiveReferences(solution, "C", new string[] { "B" });
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        public void TestSameDependencyGraphAfterOneOfMultipleReferencesRemoved()
+        {
+            // We are going to create a solution with the references:
+            //
+            // A -> B -> C -> D
+            // B -> C
+            //
+            // This solution has multiple references from B->C. We will remove one reference from B->C and verify that
+            // the project dependency graph in the solution state did not change (reference equality).
+            //
+            // We then remove the second reference, and verify that the dependency graph does change.
+
+            var solution = CreateSolutionFromReferenceMap("A:B B:C,C C:D D");
+
+            VerifyDirectReferences(solution, "A", new string[] { "B" });
+            VerifyDirectReferences(solution, "B", new string[] { "C" });
+            VerifyDirectReferences(solution, "C", new string[] { "D" });
+            VerifyDirectReferences(solution, "D", new string[] { });
+
+            VerifyTransitiveReferences(solution, "A", new string[] { "B", "C", "D" });
+            VerifyTransitiveReferences(solution, "B", new string[] { "C", "D" });
+            VerifyTransitiveReferences(solution, "C", new string[] { "D" });
+            VerifyTransitiveReferences(solution, "D", new string[] { });
+
+            VerifyDirectReverseReferences(solution, "A", new string[] { });
+            VerifyDirectReverseReferences(solution, "B", new string[] { "A" });
+            VerifyDirectReverseReferences(solution, "C", new string[] { "B" });
+            VerifyDirectReverseReferences(solution, "D", new string[] { "C" });
+
+            VerifyReverseTransitiveReferences(solution, "A", new string[] { });
+            VerifyReverseTransitiveReferences(solution, "B", new string[] { "A" });
+            VerifyReverseTransitiveReferences(solution, "C", new string[] { "A", "B" });
+            VerifyReverseTransitiveReferences(solution, "D", new string[] { "A", "B", "C" });
+
+            var dependencyGraph = solution.State.GetProjectDependencyGraph();
+            Assert.NotNull(dependencyGraph);
+
+            var b = solution.GetProjectsByName("B").Single();
+            var firstBToC = b.ProjectReferences.First(reference => reference.ProjectId.DebugName == "C");
```

Comment:
Didn't want to capture c's Id like in other tests?

Reply:
I wrote them in the other order. 😄 Will update this one.